GitUpKit/Core/GCIndex.m

```diff
@@ -485,11 +485,21 @@ - (BOOL)resetLinesInFile:(NSString*)path index:(GCIndex*)index toCommit:(GCCommi
 }
 
 - (BOOL)checkoutFileToWorkingDirectory:(NSString*)path fromIndex:(GCIndex*)index error:(NSError**)error {
+  return [self checkoutFilesToWorkingDirectory:@[ path ]
+                                     fromIndex:index
+                                         error:error];
+}
+
+- (BOOL)checkoutFilesToWorkingDirectory:(NSArray<NSString*>*)paths fromIndex:(GCIndex*)index error:(NSError**)error {
   git_checkout_options options = GIT_CHECKOUT_OPTIONS_INIT;
   options.checkout_strategy = GIT_CHECKOUT_FORCE | GIT_CHECKOUT_DONT_UPDATE_INDEX | GIT_CHECKOUT_DISABLE_PATHSPEC_MATCH;  // There's no reason to update the index
-  options.paths.count = 1;
-  const char* filePath = GCGitPathFromFileSystemPath(path);
-  options.paths.strings = (char**)&filePath;
+  options.paths.count = paths.count;
+  options.paths.strings = (char**)malloc((paths.count + 1) * sizeof(char*));
```

Comment:
I believe this would actually result in a memory leak. It was previously accessing the inner pointer of the string representation, no memory was being allocated and I don't believe it would be releasing it either. I also don't think there needs to be a `+ 1` here as you're giving the count information of how many `char *`s to expect so it won't be reading past the count.

Reply:
Oooh good catch. Not sure where that +1 comes from either. I removed it and added a `free` call after we call the libgit2 function. And I verified that it was being freed properly.